main/src/ca/uwaterloo/flix/language/phase/unification/RecordUnification.scala

```diff
@@ -77,7 +77,7 @@ object RecordUnification {
           Err(UnificationError.OccursCheck(tv, staticRow))
         } else {
           // Introduce a fresh type variable to represent one more level of the row.
-          val restRow2 = Type.freshVar(Kind.RecordRow, tvar.loc)
+          val restRow2 = Type.freshVar(Kind.RecordRow, tvar.loc)(tvar.sym.level, flix)
```

Comment:
Add a comment. Sure its ok to reuse the same level?

Reply:
I think it is OK since every component in a type should be the same level anyway (?)